documentation/building/linux-instructions.md

```diff
@@ -1,210 +1,57 @@
-Linux Prerequisites 
+Linux Prerequisites
 ===================
 
 These instructions will lead you through preparing to build and test the diagnostics repo on Linux. We'll start by showing how to set up your environment from scratch. In some cases, a version lldb/llvm that works the best on the distro will have to be built and installed.
 
 Toolchain Setup
 ---------------
 
-The following instructions will install the required packages. This only needs to be done once per machine. These instructions assume that you already have "sudo" installed. It is also recommended to create a github fork of the diagnostics repo and cloning that instead of https://github.com/dotnet/diagnostics.git directly.
+SOS plugin requires lldb v3.4 and above. It is recommeneded to use latest version of llvm toolchain, but any version >= 3.4 should work.
 
 To build or cross build for ARM on Windows or Linux see the instructions [here](https://github.com/dotnet/runtime/blob/main/docs/workflow/building/coreclr/cross-building.md#generating-the-rootfs) in the runtime repo. You will need to clone the runtime [repo](https://github.com/dotnet/runtime.git) and build the appropriate "rootfs" for arm or arm64 using these instructions. You only need to do this once.
 
-#### Ubuntu 14.04 ####
+Other prerequisites are:
 
-In order to get clang-3.9, llvm-3.9 and lldb-3.9, we need to add additional package sources (see [http://llvm.org/apt/](http://llvm.org/apt/) for the other Ubuntu versions not listed here):
+* cmake
+* python
+* icu
 
-    sudo apt-get update
-    sudo apt-get install wget
-    echo "deb http://llvm.org/apt/trusty/ llvm-toolchain-trusty main" | sudo tee /etc/apt/sources.list.d/llvm.list
-    echo "deb http://llvm.org/apt/trusty/ llvm-toolchain-trusty-3.9 main" | sudo tee -a /etc/apt/sources.list.d/llvm.list
-    wget -O - http://llvm.org/apt/llvm-snapshot.gpg.key | sudo apt-key add -
-    sudo apt-get update
+Examples
+--------
 
-Then install the required packages:
+Debian/Ubuntu:
 
-    sudo apt-get install cmake clang-3.9 gdb gettext git libicu-dev lldb-3.9 liblldb-3.9-dev libunwind8 llvm-3.9 make python python-lldb-3.9 tar zip
+```sh
+sudo apt install cmake clang curl gdb gettext git libicu-dev lldb liblldb-dev libunwind8 llvm make python python-lldb tar wget zip
+```
 
-The lldb 3.9 package needs a lib file symbolic link fixed:
+Alpine:
 
-    cd /usr/lib/llvm-3.9/lib
-    sudo ln -s ../../x86_64-linux-gnu/liblldb-3.9.so.1 liblldb-3.9.so.1
+```sh
+sudo apk add autoconf bash clang clang-dev cmake coreutils curl gcc gettext-dev git icu-dev krb5-dev libunwind-dev llvm make openssl openssl-dev python which
+```
 
-See the section below on how to clone, build and test the repo.
+Fedora/CentOS:
 
-#### Ubuntu 16.04 ####
+```sh
+sudo dnf install clang cmake openssl findutils gdb git libicu libunwind lldb-devel llvm-devel make python python2-lldb tar wget which zip
```

Comment:
The original doc used the `yum` installer on CentOS.  `dnf` works too?

Reply:
Yup, `dnf` is available on CentOS too.
AFAIK, in active versions of CentOS and Fedora, `dnf` is preferred over `yum`. But I will defer to @omajid or @tmds for more authentic answer.

Reply:
https://bytexd.com/dnf-vs-yum/

I think we mostly use `dnf` in the docker repro.

Reply:
CentOS/RHEL 7 doesn't have `dnf`, otherwise they are mostly interchangeable from a user's point of view.

Reply:
Thanks. I have added a note about `yum` for older versions.